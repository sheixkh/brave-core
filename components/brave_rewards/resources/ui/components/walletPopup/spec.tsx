/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import WalletPopup, { Props } from './index'
import { TestThemeProvider } from 'brave-ui/theme'

const props = {
  onClose: () => null,
  userName: 'tester'
}

describe('WalletSummary tests', () => {
  const baseComponent = (props: Props) => (
    <TestThemeProvider>
      <WalletPopup id='wallet-popup' {...props} >
        <p>hi</p>
      </WalletPopup>
    </TestThemeProvider>
  )

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent(props)
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent(props))
      const assertion = wrapper.find('#wallet-popup').length
      expect(assertion).toBe(1)
    })
  })
})
