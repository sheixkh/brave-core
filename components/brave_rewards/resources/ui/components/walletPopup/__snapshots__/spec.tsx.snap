// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletSummary tests basic tests matches the snapshot 1`] = `
<div
  className="sc-bdVaJa erdxI"
  id="wallet-popup"
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH cQpvIc"
    onClick={[Function]}
  >
    <div
      className="sc-htpNat cHmJaY"
    >
      <div
        className="sc-bxivhb mdRkC"
      >
        <span
          className="sc-EHOje enuskO"
        >
          <svg
            aria-hidden="true"
            className="sc-eerKOB eWPbRY"
            focusable="false"
            viewBox="0 0 32 32"
          >
            <path
              d="M18.6,27.4c-0.8,0.2-1.6,0.4-2.4,0.4h-0.1c-0.8,0-1.6-0.1-2.4-0.4c-0.4-0.1-0.8,0.1-0.9,0.5s0.1,0.8,0.5,0.9 c0.9,0.3,1.9,0.4,2.8,0.4h0.1c1,0,1.9-0.1,2.8-0.4c0.4-0.1,0.6-0.5,0.5-0.9C19.4,27.5,19,27.2,18.6,27.4z M24.9,9.6L24.9,9.6 L24.9,9.6C23.3,4.7,18.1,2,13.3,3.6c-2.8,0.9-5,3.1-5.9,5.9l0,0c-0.8,2.4-0.5,5.6,0.8,8.6c1.8,4.1,5.1,6.9,8,6.9h0.1 c2.9,0,6.2-2.8,8-6.9C25.4,15.2,25.7,12,24.9,9.6z M20.7,6c-1.7,0-3.3,0.6-4.5,1.8c-1.2-1.1-2.9-1.8-4.5-1.8 C14.3,4.1,17.9,4.1,20.7,6z M17.3,23.4c-0.8,0.4-1.6,0.4-2.4,0c-2.4-1.1-3-5.7-1.1-10c0.6-1.3,1.3-2.5,2.3-3.6 c1,1.1,1.7,2.3,2.3,3.6C20.3,17.7,19.8,22.3,17.3,23.4z M9.4,17.6C8.3,14.9,8,12.1,8.7,10C9,9,9.7,8.2,10.5,7.8 c1.3-0.6,2.9-0.2,4.5,1.1c-1.1,1.2-1.9,2.5-2.6,4c-1.2,2.5-1.5,5.4-1,8.1C10.7,20,10,18.8,9.4,17.6z M22.8,17.6 c-0.5,1.2-1.2,2.4-2.1,3.4c0.6-2.7,0.2-5.6-1-8.1c-0.6-1.5-1.5-2.8-2.6-4c1.6-1.3,3.3-1.7,4.5-1.1c0.9,0.5,1.6,1.3,1.8,2.2 C24.3,12.1,24,14.9,22.8,17.6z"
              fill="#49cc68"
            />
          </svg>
        </span>
        tester
      </div>
      <p>
        hi
      </p>
    </div>
  </div>
</div>
`;
